TMTumblrSDK/Core/TMSDKFunctions.m

```diff
@@ -58,21 +58,29 @@ @implementation TMSDKFunctions
 NSString *TMDictionaryToQueryString(NSDictionary *dictionary) {
     NSMutableArray *parameters = [NSMutableArray array];
     
-    void (^addParameter)(NSString *key, NSString *value) = ^(NSString *key, NSString *value) {
-        [parameters addObject:[NSString stringWithFormat:@"%@=%@", TMURLEncode(key), TMURLEncode(value)]];
-    };
-    
     for (NSString *key in [[dictionary allKeys] sortedArrayUsingSelector:@selector(caseInsensitiveCompare:)]) {
-        id value = dictionary[key];
-        
-        if ([value isKindOfClass:[NSArray class]]) {
-            for (NSString *arrayValue in (NSArray *)value)
-                addParameter(key, arrayValue);
-        } else
-            addParameter(key, value);
+        AddParameter(key, dictionary[key], parameters);
     }
     
     return [parameters componentsJoinedByString:@"&"];
 }
 
+#pragma mark - private
+
+void AddParameter(NSString *key, id value, NSMutableArray *parameters) {
```

Comment:
`TMAddKeyValuePairToQueryStringMutableDictionary` makes it seem like its mutating a dictionary when it looks like its adding to an array